packages/react-icons/src/utils/createSvgIcon.ts

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+// @ts-ignore
+import { css, getNativeProps, htmlElementProperties } from '@uifabric/utilities/src';
+import { classNames, MS_ICON } from 'office-ui-fabric-react/src/components/Icon/SvgIcon.styles';
+import { ISvgIconProps } from 'office-ui-fabric-react/src/components/Icon/SvgIcon.types';
+
+export type SvgIconFuncArg<TProps = ISvgIconProps> = {
+  classNames: { [iconSlot: string]: string }; // renamed from classes
+  // rtl: boolean; // how do we support this?
+  props: TProps;
+};
+
+export type SvgIconChildrenFn<TProps = ISvgIconProps> = (svgIcon: SvgIconFuncArg<TProps>) => React.ReactNode;
+
+export type SvgIconCreateFnParams<TProps> = {
+  svg: SvgIconChildrenFn<TProps & ISvgIconProps>;
+  displayName: string;
+  // handledProps?: (keyof TProps)[];
+};
+
+// TODO: SvgIconProps are different...
+const createSvgIcon = <TProps = {}>({ svg, displayName }: SvgIconCreateFnParams<TProps>) => {
+  // @ts-ignore
+  const Component: React.FC<React.HTMLAttributes<HTMLSpanElement> & TProps & ISvgIconProps> & {
+    handledProps: (keyof (TProps & ISvgIconProps))[];
+  } = props => {
+    const { className, style = {} } = props;
+
+    const nativeProps = getNativeProps<React.HTMLAttributes<HTMLElement>>(props, htmlElementProperties);
+    const containerProps = props['aria-label']
+      ? {}
+      : {
+          role: 'presentation',
+          ['aria-hidden']: true,
+        };
+
+    return React.createElement(
+      'span',
+      {
+        ...containerProps,
+        ...nativeProps,
+        className: css(MS_ICON, classNames.root, className),
```

Comment:
What's the reasoning for using MS_ICON on the new icons?

Reply:
I thought it's nice to have some common classname for all icons. Not sure if this is the right name for it, but we can change it..

Reply:
If possible, I'd let the author bring them for the new icons and keep them as vanilla as possible. We made the mistake of using ms-Icon for the existing Icon component and have seen a number of issues where developers do things like:

```css
.ms-Icon { font-size: 22px !important; }
```

We can certainly add them later, but let's try without for now.

Reply:
Will remove it!